### PR TITLE
Fix critters and monkeys not being able to use intent switching items

### LINKED ISSUE #10941: Item intent switch trigger code only works for humans
### Labels

[BUG]

### Description

Basically stun batons don't flip while held by something that isn't a human (ie critters)

### Steps to reproduce

1. Be flockdrone
2. Pick up batong and switch to harm intent
3. Batong does not flip but behaves as if it is

### Expected Behavior

I expected intent switch triggers to happen for all mobs that use intents.

### Additional Information & Screenshots

_No response_

### PR DESCRIPTION
**Pull request by FlameArrow57**
[PLAYER ACTIONS][BUG]

## About the PR 
This PR fixes items that use intent switching, like batons, not working for critters and NPC monkeys

## Why's this needed? 
Bug fix, fixes #10941

### CODE REVIEW

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Missing the checks for clicking on the HUD (unless that calls hotkey for some reason?)

**Comment by FlameArrow57** [comment]
It's handled in HUD code

File: code/mob/living/carbon/human.dm [hunk 1]
```diff
@@ -3269,6 +3265,10 @@
 	if (src.juggling())
 		src.drop_juggle()
 
+/mob/living/carbon/human/set_a_intent(intent)
+	..()
+	if(src.equipped()?.item_function_flags & USE_INTENT_SWITCH_TRIGGER)
+		src.equipped().intent_switch_trigger(src)
```
Comment (TobleroneSwordfish) [inline code/mob/living/carbon/human.dm:3271]: if these procs are exactly the same on both children types, why not move to the parent?

Comment (FlameArrow57) [reply code/mob/living/carbon/human.dm:3271]: Figured slight performance, don't know if it matters, since other mobs don't need to be checking for this

Comment (TobleroneSwordfish) [reply code/mob/living/carbon/human.dm:3271]: I think it's safe to assume if a mob is calling `set_a_intent` then it wants intent things to work?

Comment (FlameArrow57) [reply code/mob/living/carbon/human.dm:3271]: Sure

**Pull request merged**